scripts/zones/AlTaieu/mobs/Ulxzomit.lua

```diff
@@ -3,75 +3,83 @@
 --  Mob: Ul'xzomit
 -----------------------------------
 require("scripts/globals/follow")
+require("scripts/globals/status")
 -----------------------------------
 local entity = {}
 
-entity.onMobSpawn = function(mob)
+function entity.onMobSpawn(mob)
+    mob:setMobMod(xi.mobMod.ROAM_COOL, 0)
+    mob:setMobMod(xi.mobMod.ROAM_DISTANCE, 50)
+
     local mobId = mob:getID()
-    local x = mob:getXPos()
-    local y = mob:getYPos()
-    local z = mob:getZPos()
-    local r = mob:getRotPos()
+    local numFollowers = mob:getMobMod(xi.mobMod.LEADER)
 
-    if mob:getMobMod(xi.mobMod.LEADER) > 0 then
-        for i = 1, mob:getMobMod(xi.mobMod.LEADER) do
-            local followerId = mobId + i
-            local follower = GetMobByID(followerId)
-
-            if not follower:isSpawned() then
-                local newX = x + math.random(-2, 2)
-                local newY = y
-                local newZ = z + math.random(-2, 2)
-
-                follower:setSpawn(newX, newY, newZ, r)
-                follower:spawn()
-                follower:setMobFlags(1153)
+    if numFollowers == 0 then
+        -- Baby Ul'xzomit
+        mob:setMobFlags(1153)
+
+        for i = 1, 2 do
+            local leader = GetMobByID(mobId - i)
+
+            if leader and leader:getMobMod(xi.mobMod.LEADER) >= i then
+                mob:setLocalVar("leaderID", leader:getID())
+
+                if leader:isSpawned() then
+                    xi.follow.follow(mob, leader)
+                end
+
+                return
             end
+        end
+    else
+        -- Mother Ul'xzomit
+        for i = 1, numFollowers do
+            local follower = GetMobByID(mobId + i)
 
-            xi.follow.follow(follower, mob)
+            if follower and follower:isSpawned() then
+                xi.follow.follow(follower, mob)
+            end
         end
     end
 end
 
-entity.onMobEngaged = function(mob, target)
-    local followers = xi.follow.getFollowers(mob)
+function entity.onMobFight(mob, target)
+end
 
-    if not followers then
-        return
-    end
+function entity.onMobEngaged(mob, target)
+end
```

Comment:
This function is empty, so it isn't really needed. Pls remove.

Same with any other empty functions, except the main 4 basic ones (spawn, death, despawn, etc)